### ISSUE TITLE
Split the Client in a Client and an Entity Repository

### ISSUE THREAD

**Issue by leroy0211** [mapado/rest-client-sdk#15]
Wouldn't it be nicer to have a default Client which handles the Conversion and persisting, and a separate class like the Doctrine `EntityRepository` to handle lookups?

If we want to load a specific filter, we can create an EntityRepository and add the specific methods to load something specific. Next add the EntityRepository to the Entity.

If we would want to persist or convert in a different way, we extend the Client and ~~add it to the Entity~~  tell the sdkClient to use a different client

**Comment by jdeniau** [comment]
Can you write some pseudo-code to see exactly how you see it ? 

I think this is a really good idea because the default client is kind of heavy.

**Comment by leroy0211** [comment]
Let me see. Something like this (comments are in the code)


```php
<?php

use Mapado\RestClientSdk\Mapping\Annotations as Rest;

/**
 * This way no client (or repository) means the default Client/Repository will be added
 * @Rest\Entity(key="test")
 */
class Cart
{
    /**
     * @Rest\Id()
     */
    protected $id;
}


$cartClient = $sdkClient->getClient(); // this contains the default client (a parameter can give back another client)
$repository = $cartClient->getRepository('carts'); // the client can give back the default Client/Repository

$cart = $repository->find(1); // with a repository you can lookup information


// A custom repository
class CartRepository extends Repository
{
    public function findByUser(User $user)
    {
        return $this->find($user->getId()); // find is defined in the repository (currently in your AbstractClient)
    }
}

// the Entity would look like
/**
 * @Rest\Entity(repository="namespace\to\CartRepository")
 * or with a new annotation like
 * @Rest\EntityRepository("namespace\to\CartRepository")
 */
class Cart
{
    /**
     * @Rest\Id()
     */
    protected $id;
}

// now the code would look exactly the same

$cartClient = $sdkClient->getClient(); // this contains the default client (a parameter can give back another client)
$repository = $cartClient->getRepository('carts'); // the client will now give the CartRepository back instead of the default Repository

$cart = $repository->findByUser($app->getUser()); // with a repository you can lookup information
```

**Comment by jdeniau** [comment]
In fact, you can manage multiple endpoints too when using the bundle: 

You can configure something like this:
```yaml
# app/config/config.yml
mapado_rest_client_sdk:
    entity_managers:
        foo:
            server_url: 'http://foo.com:8080'
            mappings:
                prefix: /v1
                dir: '%kernel.root_dir%/../src/Foo/Entity/'
        bar:
            server_url: 'http://bar.com'
            mappings:
                dir: '%kernel.root_dir%/../src/Bar/Entity/'
```

You will have too clients: `mapado.rest_client_sdk.foo` and `mapado.rest_client_sdk.bar`, which is pretty similar to the doctrine differents entity managers. (in fact the doctrine bundle allows you to call `doctrine.orm.foo_entity_manager` and `doctrine.orm.bar_entity_manager`).

You have a point though about the component, which does not have a main container.

**Comment by jdeniau** [comment]
Fixed in #21
